datafusion/src/physical_optimizer/pruning.rs

```diff
@@ -129,12 +130,14 @@ impl PruningPredicate {
             .collect::<Vec<_>>();
         let stat_schema = Schema::new(stat_fields);
         let stat_dfschema = DFSchema::try_from(stat_schema.clone())?;
-        let execution_context_state = ExecutionContextState::new();
-        let predicate_expr = DefaultPhysicalPlanner::default().create_physical_expr(
+
+        // TODO allow these properties to be passed in
```

Comment:
Here is reason 2 for this change: again we save  a whole new `ctx_state` and `planner` for each potential constant evaluation 🎉